myceli/tests/utils/mod.rs

```diff
@@ -0,0 +1,113 @@
+use super::*;
+
+use anyhow::Result;
+use assert_fs::fixture::ChildPath;
+use assert_fs::{fixture::FileWriteBin, fixture::PathChild, TempDir};
+use blake2::{Blake2s256, Digest};
+use file_hashing::get_hash_file;
+use messages::{MessageChunker, SimpleChunker};
+use myceli::listener::Listener;
+use rand::{thread_rng, Rng, RngCore};
+use std::path::PathBuf;
+use std::thread;
+use tokio::net::UdpSocket;
+use tokio::time::sleep;
+
+pub struct TestListener {
+    pub listen_addr: String,
+    pub test_dir: TempDir,
+}
+
+impl TestListener {
+    pub fn new() -> TestListener {
+        let test_dir = TempDir::new().unwrap();
+        let mut rng = thread_rng();
+        let port_num = rng.gen_range(6000..9000);
+        let listen_addr = format!("127.0.0.1:{port_num}");
+
+        return TestListener {
+            listen_addr,
+            test_dir,
+        };
+    }
+
+    pub async fn start(&self) -> Result<()> {
+        let thread_listen_addr = self.listen_addr.to_owned();
+        let thread_db_path = self.test_dir.child("storage.db");
```

Comment:
The names look the same, but not sure, would it be beneficial to use a separate DB for testing?

Reply:
The generated `self.test_dir` is different temporary dir for each `TestListener`, so the DBs will end up as separate.